handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -407,31 +454,78 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         ctx.flush();
     }
 
-    private static final class AsyncMappingAdapter implements AsyncMapping<String, SslContext> {
+    private static final class AsyncMappingAdapter implements SslContextMapping {
+        private final AsyncMapping<? super String, ? extends SslContext> mapping;
+
+        private AsyncMappingAdapter(AsyncMapping<? super String, ? extends SslContext> mapping) {
+            this.mapping = ObjectUtil.checkNotNull(mapping, "mapping");
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Future<Supplier<? extends SslContext>> map(String input,
+                final Promise<Supplier<? extends SslContext>> promise) {
+
+            Future<SslContext> future = ((AsyncMapping<String, SslContext>) mapping).map(input,
+                    ImmediateEventExecutor.INSTANCE.<SslContext>newPromise());
+
+            if (future.isDone()) {
+                onOperationComplete(future, promise);
+            } else {
+                future.addListener(new FutureListener<SslContext>() {
+                    @Override
+                    public void operationComplete(Future<SslContext> future) {
+                        try {
+                            onOperationComplete(future, promise);
+                        } catch (Exception err) {
+                            promise.setFailure(err);
+                        }
+                    }
+                });
+            }
+
+            return promise;
+        }
+
+        private static void onOperationComplete(Future<? extends SslContext> future,
+                Promise<Supplier<? extends SslContext>> promise) {
```

Comment:
What's the convention here? Most code seems to use "Indent continuation" of 2x the normal indention. You want it to be block formatted?

```java
private static void onOperationComplete(Future<? extends SslContext> future,
                                        Promise<Supplier<? extends SslContext>> promise) {

}
```